apps/frontend/app/(client)/(main)/contest/[contestId]/@tabs/_components/RenderProblemList.tsx

```diff
@@ -0,0 +1,59 @@
+import { DataTable } from '@/app/(client)/(main)/_components/DataTable'
+import type { ProblemDataTop } from '../page'
+import { RenderProblemColumn } from './RenderProblemColumn'
+
+export function RenderProblemList(
+  state: string,
+  isRegistered: boolean,
+  problemData: ProblemDataTop
+) {
+  if (state === 'Upcoming') {
+    return (
+      <div className="h-[608px] w-[1208px] place-content-center rounded-2xl bg-[#d9d9d940]">
+        <div className="text-center text-xl font-semibold text-[#000000]">
+          {`Contest Hasn't Started`}
+        </div>
+        <div className="text-center text-base font-normal text-[#00000080]">
+          The problem list will be released after the start of the contest
+        </div>
+      </div>
+    )
+  } else if (state === 'Ongoing' && !isRegistered) {
+    return (
+      <div className="h-[608px] w-[1208px] place-content-center rounded-2xl bg-[#d9d9d940]">
+        <div className="text-center text-xl font-semibold text-[#000000]">
+          Please Register for The Contest First!
+        </div>
+        <div className="text-center text-base font-normal text-[#00000080]">
+          The problem list only be revealed to contest participants
+        </div>
+      </div>
+    )
+  } else {
+    const transformedProblemData = problemData.data.map((problem) => {
+      return {
+        ...problem,
+        id: `/problem/${problem.id}`
+      }
+    })
+
+    return (
+      <div>
+        {problemData.total === 0 ? (
+          <div>No result.</div>
+        ) : (
+          <DataTable
+            data={transformedProblemData}
+            columns={RenderProblemColumn}
+            headerStyle={{
+              order: 'text-[#8A8A8A] w-[10px] p-0 md:w-[64px] px-4',
```

Comment:
w-16 과 같이 쓸 수 있는 것들은 찾아서 고쳐도 좋을 것 같아요!
```suggestion
              order: 'text-[#8A8A8A] w-[10px] p-0 md:w-16 px-4',
```

Reply:
수정했습니다~~